public/app/features/dashboard-scene/sharing/ShareButton/utils.ts

```diff
@@ -1,16 +1,50 @@
 import { VizPanel } from '@grafana/scenes';
-import { createAndCopyDashboardShortLink } from 'app/core/utils/shortLinks';
+import { createAndCopyShareDashboardLink } from 'app/core/utils/shortLinks';
 import { getTrackingSource } from 'app/features/dashboard/components/ShareModal/utils';
 
+import store from '../../../../core/store';
 import { DashboardScene } from '../../scene/DashboardScene';
 import { DashboardInteractions } from '../../utils/interactions';
 
+export type ShareLinkConfiguration = {
+  useAbsoluteTimeRange: boolean;
+  useShortUrl: boolean;
+  theme: string;
+};
+
+export const DEFAULT_SHARE_LINK_CONFIGURATION: ShareLinkConfiguration = {
+  useAbsoluteTimeRange: true,
+  useShortUrl: true,
+  theme: 'current',
+};
+
 export const buildShareUrl = async (dashboard: DashboardScene, panel?: VizPanel) => {
+  const { useAbsoluteTimeRange, useShortUrl, theme } =
+    getShareLinkConfigurationFromStorage() || DEFAULT_SHARE_LINK_CONFIGURATION;
   DashboardInteractions.shareLinkCopied({
-    currentTimeRange: true,
-    theme: 'current',
-    shortenURL: true,
+    currentTimeRange: useAbsoluteTimeRange,
+    theme,
+    shortenURL: useShortUrl,
     shareResource: getTrackingSource(panel?.getRef()),
   });
-  return await createAndCopyDashboardShortLink(dashboard, { useAbsoluteTimeRange: true, theme: 'current' });
+  return await createAndCopyShareDashboardLink(dashboard, {
+    useAbsoluteTimeRange,
+    theme,
+    useShortUrl,
+  });
 };
+
+type ShareLinkStoreConfiguration = ShareLinkConfiguration | undefined;
+const SHARE_LINK_CONFIGURATION = 'grafana.link.shareConfiguration';
+// Function that returns share link configuration from local storage
+export function getShareLinkConfigurationFromStorage(): ShareLinkStoreConfiguration | undefined {
```

Comment:
Is this a standard pattern to return `undefined` as an expected value?
Wouldn't it be better to return the default value from the `getShareLinkConfigurationFromStorage` function directly?
It will simplify the `undefined` managing code in several places.

Reply:
We could change it. I would change the signature, though. 
`getShareLinkConfiguration` or `getShareLinkConfigurationFromStorageOrDefault`